src/ucp/core/ucp_request.inl

```diff
@@ -352,22 +352,51 @@ ucp_request_send_buffer_reg(ucp_request_t *req, ucp_md_map_t md_map,
                                   req->send.mem_type, req, uct_flags);
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_request_send_buffer_reg_lane_check(ucp_request_t *req, ucp_lane_index_t lane,
+                                       ucp_md_map_t prev_md_map, unsigned uct_flags)
+{
+    ucp_md_map_t md_map;
+
+    if (ucp_ep_md_attr(req->send.ep,
```

Comment:
maybe 
if (!ucp_ep_md_attr(....) & NEED_MEMH) {
  return UCS_OK;
}
...

Reply:
yes, it's better - done